### DISCUSSION TITLE
auth_on_register_m5 works but other auth_webhooks never fire

### DISCUSSION THREAD

**Discussion by thinkwelltwd** [vernemq/vernemq#2071] [Q&A] [answered]
I'm just getting started with VerneMQ and I'm trying to get auth_*_m5 webhooks configured. The only one works as I'd expect is `auth_on_register_m5` .  

Other auth hooks such as `auth_on_publish_m5` & `auth_on_subscribe_m5` never hit my endpoints. Nor do I see any log lines of interest in `console.log`

Other webhooks such as `on_deliver_m5` & `on_subscribe_m5` work just fine.

Here's my webhook config.

```
plugins.vmq_webhooks = on

vmq_webhooks.ccd_register.hook = auth_on_register_m5
vmq_webhooks.ccd_register.endpoint = http://localhost:1500/vmq/register/

vmq_webhooks.ccd_publish.hook = auth_on_publish_m5
vmq_webhooks.ccd_publish.endpoint = http://localhost:1500/vmq/publish/

vmq_webhooks.ccd_subscribe.hook = auth_on_subscribe_m5
vmq_webhooks.ccd_subscribe.endpoint = http://localhost:1500/vmq/subscribe/
```

Other config file settings that may or may not be relevant:

```
allow_anonymous = off
listener.ssl.require_certificate = on
plugins.vmq_acl = on
plugins.vmq_diversity = off
```

VerneMQ was built from source on CentOS 7 machine.

```
# /opt/vernemq/bin/vernemq version
1.12.6.2-2-ge33f7ab
```

Any suggestions about what to look for? Thanks!!

**Comment by ioolkos** [comment, marked as answer]
@thinkwelltwd Your Publish/Subscribe hooks are not triggered because you have activated the vmq_acl plugin which takes precedence here.

The Publish and Subscribe hooks  are related to authorization.
The Register hook is related to authentication.

It's a little confusing that the hook names start with "auth_", meaning authentication or authorization in different cases.

**Comment by thinkwelltwd** [comment]
>  you have activated the vmq_acl plugin which takes precedence here.

Well hubba hubba  :man_facepalming: it was as simple as that; It's working now. :man_cartwheeling: 

Thanks a million! 

> It's a little confusing that the hook names start with "auth_", 

Yes it is; I coulda shoulda thought of that though. :expressionless:
